### DISCUSSION TITLE
Connection aborted issues

### DISCUSSION THREAD

**Discussion by prithiviraj-manavalan-zoomrx** [sqlalchemy/sqlalchemy#12199] [Usage Questions]
Bug Report

I am encountering an issue while using the following setup:

SQLAlchemy Version: 2.0.36
FastAPI Version: 0.111.1
When attempting to establish more than three connections to a MySQL database, the connection is aborted by the server. Below is the relevant log entry from MySQL:
`
2024-12-20T08:19:07.943764Z 15 [Note] [MY-010914] [Server] Aborted connection 15 to db: '<db_name>' user: '<user_name>' host: '<host_name>' (The client was disconnected by the server because of inactivity).`
Additionally, the error returned by the server is:
`Packet sequence number wrong - got 102 expected 1`

Please let me know if additional information or logs are required.

**Comment by gordthompson** [comment]
Have you seen the documentation here?

[Dealing with Disconnects](https://docs.sqlalchemy.org/en/20/core/pooling.html#dealing-with-disconnects)

**Reply by prithiviraj-manavalan-zoomrx** [reply]
@gordthompson  I tried pool_pre_ping as well still the issue persists

**Reply by CaselIT** [reply]
the server error suggests concurrent use of the connection, either by threads or task.

can't really say much more without an example of your usage
